### ISSUE TITLE
v0.38.0 (Windows): Invalid device selection when algo is not supported (NexaPow on RX580)

### ISSUE THREAD

**Issue by UselessGuru** [andru-kun/wildrig-multi#255]
```C:\...\Bin\Wildrig-v0.38.0>print-devices
PLATFORMS:
#0: NVIDIA Corporation
#1: Advanced Micro Devices, Inc.

DEVICES:
GPU #0: Radeon RX 580 Series    (busID: 1, arch: Ellesmere, cu: 36, mem: 8192Mb)
**GPU #1: AMD Radeon RX 5700 XT   (busID: 9, arch: gfx1010:xnack-, cu: 20, mem: 8176Mb)**
GPU #2: NVIDIA GeForce GTX 1660 SUPER   (busID: 11, arch: sm75, cu: 22, mem: 6143Mb)
GPU #3: NVIDIA GeForce GT 1030  (busID: 12, arch: sm61, cu: 3, mem: 2047Mb)
GPU #4: NVIDIA GeForce GTX 750 Ti       (busID: 16, arch: sm50, cu: 5, mem: 2047Mb)
Press any key to continue . . .
```

**--opencl-devices 1** is AMD RX5700, but instead miner prints warning & attempts to use RX580:
```c:\...\Bin\Wildrig-v0.38.0>wildrig.exe --algo nexapow --api-port 4001 --url stratum+tcp://nexapow.eu.mine.zergpool.com:3004 --user ******************************************** --pass c=BTC,mc=NEXA,ID=******************,pl=0 --multiple-instance --opencl-platforms 1 **--opencl-devices 1**

 ###################################
 #################  ################
 ################    ############### * ABOUT:    WildRig/0.38.0
 ####   #########    #########   ### * CPU:      Intel(R) Core(TM) i5-8600K CPU @ 3.60GHz
 ####     #######    #######     ### * ALGO:     nexapow
 ####   #  ######    ######      ### * POOL #1:  nexapow.eu.mine.zergpool.com:3004
 ####   ##  #####    #####  ##   ### * API PORT: 4001
 ####   ##   #####  #####   ##   ### * COMMANDS: hashrate, statistics
 ####   ##   #####  #####  ###   ###
 ####   ###  #####  #####  ###   ###   devfee: 2%
 ####   ###  ####    ####  ###   ###   send-stale: no
 ####   ###   ###    ###   ###   ###   watchdog: disabled
 ####   ##    ##      ##    ##   ###
 ####      #####  ##  #####      ###
 ####   #   ###  ####  ###   #   ###
 ####   ##       ####       ##   ###
 ####   #####  #      #  #####   ###
 ####   #####  ###  ###  #####   ###
 ####                            ###
 ############   ##  ##   ###########
 #############   #  #   ############
  #############        #############
      ###########    ###########
          ##################

[09:57:27] Radeon RX 580 Series [ellesmere] is unsupported for this algo, use WildRig 0.35.2... skip
[09:57:37] initializing GPUs...
[09:57:37] Failed to find any device
[09:57:38] Failed to start OpenCL threads
```

Here I use **--opencl-devices 0** which is RX580, but then RX5700 is selected:

```c:\...\Bin\Wildrig-v0.38.0>wildrig.exe --algo nexapow --api-port 4001 --url stratum+tcp://nexapow.eu.mine.zergpool.com:3004 --user ******************************************** --pass c=BTC,mc=NEXA,ID=******************,pl=0 --multiple-instance --opencl-platforms 1 **--opencl-devices 0**

 ###################################
 #################  ################
 ################    ############### * ABOUT:    WildRig/0.38.0
 ####   #########    #########   ### * CPU:      Intel(R) Core(TM) i5-8600K CPU @ 3.60GHz
 ####     #######    #######     ### * ALGO:     nexapow
 ####   #  ######    ######      ### * POOL #1:  nexapow.eu.mine.zergpool.com:3004
 ####   ##  #####    #####  ##   ### * API PORT: 4001
 ####   ##   #####  #####   ##   ### * COMMANDS: hashrate, statistics
 ####   ##   #####  #####  ###   ###
 ####   ###  #####  #####  ###   ###   devfee: 2%
 ####   ###  ####    ####  ###   ###   send-stale: no
 ####   ###   ###    ###   ###   ###   watchdog: disabled
 ####   ##    ##      ##    ##   ###
 ####      #####  ##  #####      ###
 ####   #   ###  ####  ###   #   ###
 ####   ##       ####       ##   ###
 ####   #####  #      #  #####   ###
 ####   #####  ###  ###  #####   ###
 ####                            ###
 ############   ##  ##   ###########
 #############   #  #   ############
  #############        #############
      ###########    ###########
          ##################

[09:57:46] Radeon RX 580 Series [ellesmere] is unsupported for this algo, use WildRig 0.35.2... skip
[09:57:56] initializing GPUs...
[09:57:56] GPU #0: AMD Radeon RX 5700 XT [busID: 9] [arch: gfx1010]
[09:57:56] threads: 1, intensity: 22, worksize: 0/256, cu: 40, mem: 8176Mb
[09:57:57] use pool nexapow.eu.mine.zergpool.com:3004 141.95.55.97
[09:57:57] Stratum set raw difficulty to 0.0500
[09:57:57] new job from nexapow.eu.mine.zergpool.com:3004 diff 214.75M/1.12P
[09:57:57] block: 399,503       job target: 0x00000013ffec0000
[09:57:57] GPU #0: start precompute
[09:57:58] Ctrl+C received
[09:57:58] exiting...
[09:57:58] GPU #0: shutting down...
```

**Comment by andru-kun** [comment]
dang, I knew I'm missing something xD Thanks, will take a look on this

**Comment by UselessGuru** [comment]
Fixed in 0.38.1
